content/zh/docs/concepts/services-networking/service.md

```diff
@@ -894,7 +894,7 @@ This flag takes a comma-delimited list of IP blocks (e.g. 10.0.0.0/8, 192.0.2.0/
 
 如果将 `type` 字段设置为 `NodePort`，则 Kubernetes 控制平面将在 `--service-node-port-range` 标志指定的范围内分配端口（默认值：30000-32767）。
 每个节点将那个端口（每个节点上的相同端口号）代理到您的服务中。
-您的服务在其 `.spec.ports[*].nodePort` 字段中要求分配的端口。
+您的服务在其 `.spec.ports[*].nodePort` 字段中报告已分配的端口。
```

Comment:
spec 中的内容是请求的内容，不是被分配的内容。

Reply:
也许说的是describe的结果？

就像这样：
➜  ~ minikube start
😄  Darwin 10.15.5 上的 minikube v1.14.2
✨  根据现有的配置文件使用 hyperkit 驱动程序
👍  Starting control plane node minikube in cluster minikube
🏃  Updating the running hyperkit "minikube" VM ...
❗  This VM is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  正在 Docker 19.03.12 中准备 Kubernetes v1.19.2…
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" by default
➜  ~ kubectl create deployment hello-minikube --image=k8s.gcr.io/echoserver:1.10
deployment.apps/hello-minikube created
➜  ~ kubectl expose deployment hello-minikube --type=NodePort --port=8080
service/hello-minikube exposed
➜  ~ kubectl get svc -o custom-columns=spec:spec hello-minikube
spec
map[clusterIP:10.101.146.114 externalTrafficPolicy:Cluster ports:[map[**nodePort:32735** port:8080 protocol:TCP targetPort:8080]] selector:map[app:hello-minikube] sessionAffinity:None type:NodePort]

Reply:
No. You cannot specify nodeport using `kubectl expose` yet. See the following output:

```
kubectl explain service.spec.ports
KIND:     Service
VERSION:  v1

RESOURCE: ports <[]Object>

DESCRIPTION:
     The list of ports that are exposed by this service. More info:
     https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies

     ServicePort contains information on service's port.

FIELDS:
   appProtocol	<string>
     The application protocol for this port. This field follows standard
     Kubernetes label syntax. Un-prefixed names are reserved for IANA standard
     service names (as per RFC-6335 and
     http://www.iana.org/assignments/service-names). Non-standard protocols
     should use prefixed names such as mycompany.com/my-custom-protocol. This is
     a beta field that is guarded by the ServiceAppProtocol feature gate and
     enabled by default.

   name	<string>
     The name of this port within the service. This must be a DNS_LABEL. All
     ports within a ServiceSpec must have unique names. When considering the
     endpoints for a Service, this must match the 'name' field in the
     EndpointPort. Optional if only one ServicePort is defined on this service.

   nodePort	<integer>
     The port on each node on which this service is exposed when type=NodePort
     or LoadBalancer. Usually assigned by the system. If specified, it will be
     allocated to the service if unused or else creation of the service will
     fail. Default is to auto-allocate a port if the ServiceType of this Service
     requires one. More info:
     https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport

   port	<integer> -required-
     The port that will be exposed by this service.

   protocol	<string>
     The IP protocol for this port. Supports "TCP", "UDP", and "SCTP". Default
     is TCP.

   targetPort	<string>
     Number or name of the port to access on the pods targeted by the service.
     Number must be in the range 1 to 65535. Name must be an IANA_SVC_NAME. If
     this is a string, it will be looked up as a named port in the target Pod's
     container ports. If this is not specified, the value of the 'port' field is
     used (an identity map). This field is ignored for services with
     clusterIP=None, and should be omitted or set equal to the 'port' field.
     More info:
     https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
```

`kubectl expose` can handle `port` and `targetPort`, not `nodePort`. Usually you will leave it for the system to decide its value. But you CAN specify a fixed `nodePort` in certain scenarios. If you DO, you have to make sure that node port is available for use.

Reply:
@tengqm 你说的对，但我纠结的不是`nodePort `字段的功能，我纠结的是文档中的这句话，翻译的很奇怪。

或许我们可以改成这样：
您的服务可以通过 `.spec.ports[*].nodePort` 字段指定分配的端口。

当然，如果你还是觉得原文更好，那就请关闭这个issue吧

Reply:
> 您的服务可以通过 .spec.ports[*].nodePort 字段指定分配的端口。

> 您的服务在其 `.spec.ports[*].nodePort` 字段中要求分配的端口。

所以这不是一个正确性的问题，而是一个中文表述的问题。